### DISCUSSION TITLE
ModbusRTU server and a ModbusTCP client NOT bridge

### DISCUSSION THREAD

**Discussion by JesperSOGT** [eModbus/eModbus#275] [Q&A]
Hi, I have a Modbus RTU server running and now I need to read some data from a Modbus TCP device. However, when I add the #include <ModbusClientTCP.h> to my program, I get a lot of errors from the RTU library. Is it not possible to run both models at the same time?.

**Comment by Miq1** [comment]
That should be possible without any problem - I have several sketches doing that.

Can you please post your sketch and the errors you get here?

**Comment by JesperSOGT** [comment]
I updated the library and adjusted it according to the new documentation, but I still get the same error. Even when I try the example RTU server, I get errors. Could you verify that the examples can be compiled with the updates you made?
 error: 'class HardwareSerial' has no member named 'setTxBufferSize'
error: expected unqualified-id before '.' token   RTUutils.prepareHardwareSerial(Serial2);

// Init Serial2 connected to the RTU Modbus
// (Fill in your data here!)
  RTUutils.prepareHardwareSerial(Serial2);
  Serial2.begin(19200, SERIAL_8N1, GPIO_NUM_17, GPIO_NUM_16);

// Register served function code worker for server 1, FC 0x03
  MBserver.registerWorker(0x01, READ_HOLD_REGISTER, &FC03);

// Start ModbusRTU background task
  MBserver.begin(Serial2);

**Reply by Miq1** [reply]
The errors are pointing to you not having a more recent arduino-esp32 core installed. Prerequisite is a core version of 2.0.x. You should update it anyway, as a multitude of changes have been integrated for all areas.

**Reply by Miq1** [reply]
Oh, btw, the second results from the "." you used to address the ``prepareHardwareSerial`` method. It is static, so you will need to write ``RTUutils::prepareHardwareSerial(Serial2);`` instead.
